syntax = "proto3";

package second.protos;
import "second/protos/anchors.proto";
import "second/protos/similarity.proto";

message ClassSetting {
  oneof anchor_generator {
    AnchorGeneratorStride anchor_generator_stride = 1;
    AnchorGeneratorRange anchor_generator_range = 2;
    NoAnchor no_anchor = 3;
  }
  RegionSimilarityCalculator region_similarity_calculator = 4;
  bool use_multi_class_nms = 5;
  bool use_rotate_nms = 6;
  int32 nms_pre_max_size = 7;
  int32 nms_post_max_size = 8;
  float nms_score_threshold = 9;
  float nms_iou_threshold = 10;
  float matched_threshold = 11;
  float unmatched_threshold = 12;
  string class_name = 13;
  repeated int64 feature_map_size = 14; // 3D zyx (DHW) size
}

message TargetAssigner {
  repeated ClassSetting class_settings = 1;
  float sample_positive_fraction = 2;
  uint32 sample_size = 3;
  bool assign_per_class = 4;
  repeated int64 nms_pre_max_sizes = 5; // this will override setting in ClassSettings if provide.
  repeated int64 nms_post_max_sizes = 6; // this will override setting in ClassSettings if provide.
  repeated int64 nms_score_thresholds = 7; // this will override setting in ClassSettings if provide.
  repeated int64 nms_iou_thresholds = 8; // this will override setting in ClassSettings if provide.
}